sdnfv/openNetVM
# NF core reassignment on termination

Adds functionality to onvm_mgr to reassign cores on NF shutdown.


## Summary:
Reallocates a NF to another core when another NF shuts down and a better core allocation is possible.
This is a part of multiple changes regarding the autoscaling and core managing improvments for ONVM, working with @ratnadeepb on these.

**Usage:**
Run onvm_mgr with 2 cores for NFs, f.e:
` ./go.sh 0,1,2 0 0x60`
Run basic monitor (or any other NF really)
`./go.sh 2`
Run basic monitor in shared more  (or any other NF really)
`./go.sh -l 0 -- -r 1 -s -- `
Run another  basic monitor in shared more  (or any other NF really)
`./go.sh -l 0 -- -r 1 -s -- `
*Then Ctrl-C the first NF (the one not running in shared mode)
The second NF should now be reallocated to the core that the first NF was running on. 


| This PR includes         |          |
| ------------------------ | -------- |
| Resolves issues          |  
| Breaking API changes     |
| Internal API changes     |
| Usability improvements   |  
| Bug fixes                |
| New functionality        | :+1: 
| New NF/onvm_mgr args     | 
| Changes to starting NFs  |  
| Dependency updates       | 
| Web stats updates        | 



## Merging notes:
 - Dependencies: None  

**TODO before merging :**



## Test Plan:
Figure out if this can break anything.


## Review: 
### Review checklist:  

- Sanity checks, assigned to @dennisafa  @rskennedy 
    - Run make in `/onvm` and `/examples` directories
    - Test the new functionality

- Code style, assigned to @dennisafa  @rskennedy 
    - Run linter
    - Check everything style related

- Code design, assigned to @dennisafa  @rskennedy 
    - Check for memory leaks
    - Check that code is reusable
    - Code is clean, functions are concise
    - Verify that edge cases properly handled

- Performance, assigned to @dennisafa  @rskennedy 
    - Run Speed Tester NF, report performance.
    - Run pktgen, report performance (not really required)

- Documentation, assigned to @dennisafa  @rskennedy 
    - Check if the new changes are well documented, in both code and READMEs
